### PR TITLE
HOTFIX: Close transactional producers in all new tests

### PR DESCRIPTION
**Pull request by apurvam**

### CODE REVIEW

**Comment by apurvam** [comment]
cc @hachikuji @ijuma

**Review by hachikuji — APPROVED** [review]
LGTM. Let's wait for the jenkins builds and see if this corrects the problem.

File: core/src/test/scala/integration/kafka/api/TransactionsTest.scala [hunk 1]
```diff
@@ -352,6 +351,7 @@ class TransactionsTest extends KafkaServerTestHarness {
       }
     } finally {
       consumer.close()
+      producer1.close()
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/TransactionsTest.scala:354]: We can do it this way for now, but one thing that we've done to make tests more robust is to use a factory method that registers the instance in a collection. During tearDown, we can then close everything.

Comment (apurvam) [reply core/src/test/scala/integration/kafka/api/TransactionsTest.scala:354]: That's an excellent suggestion. I created this JIRA to keep track of this, and will get to it after we fix the existing blockers: https://issues.apache.org/jira/browse/KAFKA-5282

**Comment by hachikuji** [comment]
I don't see any of the garbage in the test output, so it looks like this fixed the problem.

**Pull request closed without merging**